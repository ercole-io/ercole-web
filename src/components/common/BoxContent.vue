<template>
  <section :style="{ 'background-color': bgcolor }">
    <h2 class="sub-title" v-if="title" :class="{ 'border-bottom': border }">
      {{ title }}
      <slot name="customTitle" />
    </h2>
    <article
      class="box-content"
      :class="{ 'bottom-space': mbottom, 'card card-custom': hasShadow }"
    >
      <slot />
    </article>
  </section>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: null,
    },
    border: {
      type: Boolean,
      default: false,
    },
    mbottom: {
      type: Boolean,
      default: true,
    },
    bgcolor: {
      type: String,
      default: '#ffffff',
    },

    hasShadow: {
      type: Boolean,
      default: false,
    },
  },
}
</script>

<style lang="scss" scoped>
@import '@/assets/scss/_variables';
.box-content {
  padding: 0 0.5rem;
  height: 100%;
  position: relative;
}

.bottom-space {
  margin-bottom: 20px;
}

.sub-title {
  font-size: 0.9em;
  margin-bottom: 10px;
  padding: 5px;
  font-weight: 500;
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: relative;
  color: $ercole-blue;
}

.border-bottom {
  border-bottom: 1px solid #ccc;
  padding-bottom: 5px;
  margin-bottom: 15px;
}

.card-custom {
  padding: 20px 0;
}
</style>
