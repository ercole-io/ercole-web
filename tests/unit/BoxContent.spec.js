import BoxContent from '@/components/common/BoxContent'
import { mount } from '@vue/test-utils'

describe('Ercole Logo', () => {
  it('will mount the component', () => {
    const wrapper = mount(BoxContent, {
      propsData: {
        title: 'Title Test',
        border: true,
        mbottom: false,
        bgcolor: '#000000'
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
