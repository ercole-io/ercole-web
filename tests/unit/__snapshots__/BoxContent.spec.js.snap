// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ercole Logo will mount the component 1`] = `
<section style="background-color: rgb(0, 0, 0);">
  <h2 class="sub-title border-bottom">
    Title Test
  </h2>
  <article class="box-content"></article>
</section>
`;
